packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -63,11 +78,25 @@ class _TextSelectionToolbarState extends State<_TextSelectionToolbar> with Ticke
   // The key for _TextSelectionToolbarContainer.
   UniqueKey _containerKey = UniqueKey();
 
-  FlatButton _getItem(VoidCallback onPressed, String label) {
-    assert(onPressed != null);
-    return FlatButton(
-      child: Text(label),
-      onPressed: onPressed,
+  Widget _getItem(_ItemData itemData, bool isFirst, bool isLast) {
+    assert(isFirst != null);
+    assert(isLast != null);
+    return ButtonTheme.fromButtonThemeData(
+      data: ButtonTheme.of(context).copyWith(
+        height: kMinInteractiveDimension,
+        minWidth: kMinInteractiveDimension,
+      ),
+      child: FlatButton(
+        onPressed: itemData.onPressed,
+        padding: EdgeInsets.only(
+          // These values were eyeballed to match the native text selection menu
+          // on a Pixel 2 running Android 10.
+          left: 9.5 + (isFirst ? 5.0 : 0.0),
```

Comment:
With `RTL` text direction the extra horizontal padding (5.0) is going to face inward I think. Maybe use `EdgeInsetsDirectional` instead?

Reply:
I played around with this in DartPad and it seems that EdgeInsetsDirectional start is what will face inward when the text direction is RTL, so it should be correct as-is.  Let me know if I'm not thinking about this right though. https://dartpad.dartlang.org/98d77d79c6664a4f64968b899836419f

Reply:
If the direction is `RTL` then in the column the first item in the list will be placed at the right edge with extra padding on its left I think?

Reply:
In the dartpad if I do `crossAxisAlignment: CrossAxisAlignment.start` on the column and set the text direction to RTL for the column, then the buttons do align right.  In selection menu though, the position of the buttons isn't affected by text direction.  It also seems like when the menu is overflowing and some buttons end up in an overflow menu in a column, they are always aligned left.  Does that make sense or am I missing the case where it doesn't work?

Reply:
Oh sorry I thought the buttons will be put in a text direction aware container like a `Row`. Looks like `_TextSelectionToolbarContainer` doesn't care about the text direction. It appears if I turn "force RTL" on in a native android application the menu will flip. I'll file a bug if there isn't one

Reply:
Thanks!  I didn't realize the menu order flipped on native.